### PR TITLE
Migrate Material UI to v4

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]

##### Description of change

#1522 
Involved jumping through some hoops to get the breaking changes fixed.
NOTE: This breaks the course editor page, which is soon to be decommissioned.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/EntityList.tsx [hunk 1]
```diff
@@ -36,16 +35,16 @@ const styles = (theme: Theme) =>
...
+      right: theme.spacing(5)
     }
   } as StyleRules);
 
 interface EntityListProps extends WithStyles<typeof styles> {
   resultsText: React.ReactNode;
   resultsRight?: React.ReactNode;
   children: React.ReactNode;
-  create?: React.ComponentType<FabProps>;
+  create?: any;
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/components/EntityList.tsx:47]: Do you really have to use `any`? Can't use `React.ComponentType<Fab>` or something from `Fab`?

File: Source/Plugins/Core/com.equella.core/js/tsrc/theme/ThemePage.tsx [hunk 2]
```diff
@@ -275,10 +275,10 @@ class ThemePage extends React.Component<
       <div>
         <CardContent className={classes.cardContent}>
           <FormControl>
-            <Typography variant={"display1"}>
+            <Typography variant={"h4"}>
               {strings.colourschemesettings.title}
             </Typography>
-            <Grid container spacing={16}>
+            <Grid container spacing={8}>
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/theme/ThemePage.tsx:281]: Why the change?

Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/theme/ThemePage.tsx:281]: The spacing API has changed. The possible values used to be [0, 8, 16, 24, 32, 40] and now they are [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10].

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/EntityList.tsx (hunk 1 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/EntityList.tsx]: Fixed - the create prop was only used in one place, and in a really strange way. Refactored around it and removed the prop.

File: Source/Plugins/Core/com.equella.core/js/tsrc/theme/ThemePage.tsx (hunk 2 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/theme/ThemePage.tsx]: Although, this certainly highlighted that it should actually be 2, not 8.
